neuralmonkey/dataset.py

```diff
@@ -0,0 +1,572 @@
+    """Get paths to files that contain data from the dataset kwargs.
+
+    Input file for a serie named 'xxx' is specified by parameter 's_xxx'. The
+    dataset series is defined by a string with a path / list of strings with
+    paths, or a tuple whose first member is a path or a list of paths and the
+    second memeber is a reader function.
+
+    The paths can contain wildcards, which will be expanded using
+    :py:func:`glob.glob` in sorted order.
+
+    Arguments:
+        series_config: A dictionary containing the dataset keyword argument
+            specs.
+
+    Returns:
+        A dictionary which maps serie names to the paths of their input files
+        and readers..
+    """
+    keys = [k for k in list(series_config.keys()) if SERIES_SOURCE.match(k)]
+    names = [get_first_match(SERIES_SOURCE, k) for k in keys]
+
+    series_sources = {}
+    for name, key in zip(names, keys):
+        value = cast(ReaderDef, series_config[key])
+
+        if isinstance(value, tuple):
+            patterns, reader = value  # type: ignore
+        else:
+            patterns = value
+            reader = UtfPlainTextReader
+
+        if isinstance(patterns, str):
+            patterns = [patterns]
+
+        paths = _expand_patterns_flat(patterns)
+        debug("Series '{}' has the following files: {}".format(name, paths))
+
+        series_sources[name] = (paths, reader)
+
+    return series_sources
+
+
+def _get_series_outputs(series_config: SeriesConfig) -> List[OutputSpec]:
+    """Get paths to series outputs from the dataset keyword argument specs.
+
+    Output file for a series named 'xxx' is specified by parameter 's_xxx_out'
+
+    Arguments:
+        series_config: A dictionary containing the dataset keyword argument
+           specs.
+
+    Returns:
+        A dictionary which maps serie names to the paths for their output
+        files.
+    """
+    outputs = {}
+    for key, value in series_config.items():
+        matcher = SERIES_OUTPUT.match(key)
+        if matcher:
+            name = matcher.group(1)
+            if not isinstance(value, str):
+                raise ValueError(
+                    "Output path for '{}' series must be a string, was {}.".
+                    format(name, type(value)))
+            outputs[name] = cast(str, value)
+    return [(key, val, AutoWriter) for key, val in outputs.items()]
+
+
+# pylint: disable=too-many-locals
+# This is a deprecated function, no point in removing one local var from it
+def load_dataset_from_files(
+        name: str,
+        lazy: bool = False,
+        preprocessors: List[Tuple[str, str, Callable]] = None,
+        **kwargs) -> "Dataset":
+    """Load a dataset from the files specified by the provided arguments.
+
+    Paths to the data are provided in a form of dictionary.
+
+    Keyword arguments:
+        name: The name of the dataset to use. If None (default), the name will
+              be inferred from the file names.
```

Comment:
Tady to chce update docstringu. Z argumentu metody nevypliva, ze None je default + name se neinferuje s filenames, ale z nazvu objektu v configu, ne?

Reply:
na to kašli, to je deprecated funkce